framework/src/utils/ElementsIntersectedByPlane.C

```diff
@@ -0,0 +1,79 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// Moose includes
+#include "ElementsIntersectedByPlane.h"
+
+// libMesh includes
+#include "libmesh/plane.h"
+#include "libmesh/point.h"
+#include "libmesh/mesh.h"
+#include "libmesh/elem.h"
+
+#include <algorithm>
+
+namespace Moose
+{
+
+void findElementsIntersectedByPlane(const Plane & plane, const MeshBase & mesh, std::vector<Elem *> & intersected_elems)
+{
+  // Loop over all elements to find elements intersected by the plane
+  MeshBase::const_element_iterator el = mesh.elements_begin();
+  const MeshBase::const_element_iterator end_el = mesh.elements_end();
+  for (; el != end_el; ++el)
+  {
+    const Elem * elem = *el;
+    const unsigned int dim = elem->dim();
+    std::vector<bool> above_plane;
+
+    // Loop over all nodes belonging to this element and check if they are above the plane or not
+    for (unsigned int i = 0; i < elem->n_nodes(); ++i)
+    {
+      const Node node = *elem->get_node(i);
+
+      bool node_above = plane.above_surface(node);
+      above_plane.push_back(node_above);
+    }
+
+    // Elements with nodes both above and below the plane are determined to be intersected
+    if (std::find(above_plane.begin(), above_plane.end(), true)  != above_plane.end() &&
+        std::find(above_plane.begin(), above_plane.end(), false) != above_plane.end())
+      intersected_elems.push_back(const_cast<Elem *>(elem));
```

Comment:
I'd recommend that you avoid a `const_cast`. Why not pass in a `std::vector<const Elem *>` instead?